pkg/services/dashboardsnapshots/service.go

```diff
@@ -39,6 +40,14 @@ func CreateDashboardSnapshot(c *contextmodel.ReqContext, cfg dashboardsnapshot.S
 		return
 	}
 
+	uid := cmd.DashboardCreateCommand.Dashboard.GetNestedString("uid")
+
+	dashboardExists, err := svc.ValidateDashboardExists(c.Req.Context(), c.SignedInUser.GetOrgID(), uid)
+	if !dashboardExists || err != nil {
+		c.JsonApiErr(http.StatusBadRequest, "Failed to find dashboard", err)
+		return
+	}
```

Comment:
I looked at the dashboard service and it is returning a specific error when the dashboard is not found. We don't need the boolean in this case, we can do something like this: 
```suggestion
		err := svc.ValidateDashboardExists(c.Req.Context(), c.SignedInUser.GetOrgID(), uid)
	if err != nil {
		if errors.Is(err, dashboards.ErrDashboardNotFound) {
			c.JsonApiErr(http.StatusBadRequest, "Dashboard not found", err)
			return	
		}
		c.JsonApiErr(http.StatusInternalServerError, "Failed to find dashboard", err)
		return
	}
```
The idea is to execute the validation and if something happen we return the error from the dashboard service layer. If it is a specific `ErrDashboardNotFound` we can inform the user a Bad request error asking for a different dashboard. Otherwise, if it is an unexpected error, we should return the Internal server error.
WDYT?

Reply:
It makes sense to separate out the error. If that's the case, I think I need to change the InternalServiceError message to something else that doesn't sound the same